src/wasm-interpreter.h

```diff
@@ -1919,6 +1919,33 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
     }
     return Literal(int32_t(data->values.size()));
   }
+  Flow visitStringConcat(StringConcat* curr) { 
+    NOTE_ENTER("StringConcat");
+    Flow flow = visit(curr->left);
+    if (flow.breaking()) {
+      return flow;
+    }
+    auto left = flow.getSingleValue();
+    flow = visit(curr->right);
+    if (flow.breaking()) {
+      return flow;
+    }
+    auto right = flow.getSingleValue();
+    NOTE_EVAL2(left, right);
+    auto leftData = left.getGCData();
+    auto rightData = right.getGCData();
+    char array[leftData->values.size() + rightData->values.size()+1];
+    int i =0;
+    for (Literal l : leftData->values) {
+      array[i++] = l.getInteger();
+    }
+
+    for (Literal l : rightData->values) {
+      array[i++] = l.getInteger();
+    }
+    array[i] = 0;
```

Comment:
Does the spec mention adding a null? I would guess not but I didn't read it carefully just now.

Reply:
I though that `std::string(char*)` would take a `null` terminated string.  Removed. I haven't programmed in c++ since the late 90s :)

Reply:
Ah, I see now. Yeah, I'm not sure about C++ either actually :smile: , but see my comment from a few seconds ago about `makeGCData` that avoids this.